### PR TITLE
Also retry failed POST requests.

### LINKED ISSUE #144: Bodhi2Client doesn't support retries
`Bodhi1Client` support retries: it tries several times (user specified) and after last unsuccessful try it raises an exception. This unfortunately doesn't work when `Bodhi2Client` is used.

When you try to run code bellow, you can see that when `Bodhi1Client` is used it actually tries 10 times before raising. `Bodhi2Client` fails that respond cannot be parsed as JSON.

    from fedora.client.bodhi import Bodhi1Client, Bodhi2Client
    import sys
    import logging

    logging.basicConfig()
    logging.getLogger().setLevel(level=logging.DEBUG)

    if int(sys.argv[1]) == 1:
        client = Bodhi1Client(base_url='http://httpstat.us/', retries=10, debug=True)
        client.send_request('500')
    else:
        client = Bodhi2Client(base_url='http://httpstat.us/', retries=10, debug=True)
        client.send_request('500')

I check the code and it seems that `Bodhi2Client`'s parent class `OpenIdBaseClient` overrides method `send_request` from its parent class `OpenIdProxyClient`, in which the retry code is present.

### PR DESCRIPTION
**Pull request by ralphbean**
This should fix #144.

### CODE REVIEW

File: fedora/client/openidbaseclient.py [hunk 1]
```diff
@@ -170,6 +170,7 @@ def __init__(self, base_url, login_url=None, useragent=None, debug=False,
         # Also hold on to retry logic.
         # http://www.coglib.com/~icordasc/blog/2014/12/retries-in-requests.html
         server_errors = [500, 501, 502, 503, 504, 506, 507, 508, 509, 599]
+        method_whitelist = Retry.DEFAULT_METHOD_WHITELIST.union(set(['POST']))
```
Comment (pypingou) [inline fedora/client/openidbaseclient.py]: might be easier to do a ``.add()`` instead of this union, no?

**Comment by pypingou** [comment]
One stylistic comment, otherwise :+1: for me

File: fedora/client/openidbaseclient.py (hunk 1 above)
Comment (pypingou) [inline fedora/client/openidbaseclient.py]: Ok NM, seeing the comment on https://github.com/fedora-infra/python-fedora/issues/144#issuecomment-160739487 ``Retry.DEFAULT_METHOD_WHITELIST`` is a frozenset and therefore does not have a ``.add()``. Tbh, I'm even surprised it supports ``.union()`` and this makes it modifiable.

Comment (kparal) [inline fedora/client/openidbaseclient.py]: It does not modify it, it returns a new `frozenset` instance.

**Comment by kparal** [comment]
I have tested the patch and it works. Thanks @ralphbean.

**Comment by ralphbean** [comment]
This should be included in the python-fedora-0.6.4 release.

**Comment by ralphbean** [comment]
OK, they're [on their way](https://bodhi.fedoraproject.org/updates/?packages=python-fedora).

**Pull request merged**